packages/spatial/src/renderer/functions/changeRenderMode.ts

```diff
@@ -60,24 +64,20 @@ export function changeRenderMode() {
     (p) => p.name === 'RenderPass'
   ) as any
   const renderPass: RenderPass = passes ? passes[0] : undefined
-
   if (!renderPass) return
 
   switch (renderMode) {
     case RenderModes.UNLIT:
-      // See above comment
-      // iterateObject3D(Engine.instance.scene, (obj: Light) => {
-      //   if (obj.isLight && obj.visible) {
-      //     obj.userData.editor_disabled = true
-      //     obj.visible = false
-      //   }
-      // })
-      renderPass.overrideMaterial = null!
-      break
     case RenderModes.LIT:
-      renderPass.overrideMaterial = null!
-      break
     case RenderModes.SHADOW:
+      iterateEntityNode(getState(EditorState).rootEntity, (entity, index) => {
+        const groups = getComponent(entity, GroupComponent)
+        for (const group of groups) {
+          if ((group as any).isLight && !(group as any).isAmbientLight) {
+            setVisibleComponent(group.entity, renderMode !== RenderModes.UNLIT)
```

Comment:
rather than doing this here, we have two other options:
- have each light component individually react to the renderer state
- create a system just before the render step that overrides the visible component to disable the lights based on renderer state

I'd preferrably opt for the second option.

Reply:
i would prefer the second option as well, ill implement it